subcommands/waves/init.go

```diff
@@ -0,0 +1,160 @@
+The same expiration will be used when a wave is promoted.
+When set this value overrides an 'expires-days' argument.
+Example: 2020-01-01T00:00:00Z`)
+	initCmd.Flags().BoolP("dry-run", "d", false, "Don't create a wave, print it to standard output.")
+	// TODO: Add an optional offline signing arguments
+}
+
+func doInitWave(cmd *cobra.Command, args []string) {
+	factory := viper.GetString("factory")
+	name, version, tag := args[0], args[1], "master"
+	if len(args) > 2 {
+		tag = args[2]
+	}
+	intVersion, err := strconv.ParseInt(version, 10, 32)
+	subcommands.DieNotNil(err, "Version must be an integer")
+	expires := readExpiration(cmd)
+	dryRun, _ := cmd.Flags().GetBool("dry-run")
+	logrus.Debugf("Creating a wave %s for factory %s targets version %s and new tag %s expires %s",
+		name, factory, version, tag, expires.Format(time.RFC3339))
+
+	new_targets, err := api.TargetsList(factory, version)
+	subcommands.DieNotNil(err)
+	if len(new_targets) == 0 {
+		subcommands.DieNotNil(fmt.Errorf("No new targets of a given version"))
+	}
+
+	current_targets, err := api.PromotedTargetsGet(factory, tag, false)
+	subcommands.DieNotNil(err)
+
+	// A standard tuf.Targets sneaks a delegations field into a resulting JSON.  As we don't use (and
+	// don't need) that field, a workaround is to use our own simplified type.  An omitempty in that
+	// field's JSON flags doesn't help as it is a struct, not a reference.
+	//
+	// The other problem goes to an Expires field.  A default Golang format for time.Time objects has
+	// changed from RFC3339 to RFC3339Nano in some version, but this new format isn't recognized as a
+	// valid datetime format by aktualizr.  Thus, we neeed to format it ourselves.
+	type TufTargets struct {
+		Type    string `json:"_type"`
+		Expires string `json:"expires"`
+		// Expires time.Time `json:"expires"`  // incompatible
+		Version int       `json:"version"`
+		Targets tuf.Files `json:"targets"`
+		// Delegations tuf.Delegations `json:"delegations,omitempty"` // unnecessary
+	}
+	targets := TufTargets{}
+	targets.Type = tuf.TUFTypes["targets"]
+	targets.Expires = expires.Format(time.RFC3339)
+	targets.Version = int(intVersion)
+	if current_targets == nil {
+		targets.Targets = make(tuf.Files)
+	} else {
+		targets.Targets = current_targets.Signed.Targets
+		if targets.Version <= current_targets.Signed.Version {
+			subcommands.DieNotNil(fmt.Errorf(
+				"Cannot create a wave for a version lower than current promoted targets"))
+		}
+	}
+
+	// Add new targets, replacing tags in each of them
+	for name, file := range new_targets {
+		// There is a dilemma what to do if a specific target already exists in current promoted
+		// targets for a given tag.  A solution below would fail in this scenario as there is no point
+		// to start a new wave for something that is already generally available in production.
+		// Under normal circumstances a target, once created, is immutable making this approach sane.
```

Comment:
I think this is so obvious that the comment isn't that useful. The error message shown to the user is enough to guide the person reading the code.

Reply:
Yes, sure... at a time I wrote this comment a code wasn't obvious, but now it is.